internal/commands/db_remove.go

```diff
@@ -0,0 +1,194 @@
+package commands
+
+import (
+	"fmt"
+
+	"github.com/dustin/go-humanize"
+	"github.com/spf13/cobra"
+)
+
+var (
+	removeClusterName string
+	removeKPIName     string
+	removeAll         bool
+)
+
+var removeCmd = &cobra.Command{
+	Use:   "remove",
+	Short: "Delete data from the database",
+	Long: `Delete clusters, KPI metrics, or error records from the database.
+
+	WARNING: All remove operations are immediate and cannot be undone.`,
+}
+
+var removeClustersCmd = &cobra.Command{
+	Use:   "clusters",
+	Short: "Remove a cluster and all its data",
+	Long: `Delete a cluster record and all associated KPI metrics from the database.
+
+WARNING: This operation cannot be undone. All metric samples for the cluster will be permanently deleted.`,
+	Example: `  # Remove a cluster
+  kpi-collector db remove clusters --name="old-cluster"`,
+	RunE: runRemoveClusters,
+}
+
+var removeKPIsCmd = &cobra.Command{
+	Use:   "kpis",
+	Short: "Remove KPI metrics",
+	Long:  `Delete KPI metrics from the database, optionally filtered by cluster and KPI name.`,
+	Example: `  # Remove all KPIs from a cluster
+  kpi-collector db remove kpis --cluster-name="mycluster1"
+  
+  # Remove specific KPI from a cluster
+  kpi-collector db remove kpis --cluster-name="mycluster1" --name="cpu-system"`,
+	RunE: runRemoveKPIs,
+}
+
+var removeErrorsCmd = &cobra.Command{
+	Use:   "errors",
+	Short: "Clear error counts",
+	Long:  `Reset error counts for KPI queries.`,
+	Example: `  # Clear errors for a specific KPI
+  kpi-collector db remove errors --name="cpu-system"
+  
+  # Clear all errors
+  kpi-collector db remove errors --all`,
+	RunE: runRemoveErrors,
+}
+
+func init() {
+	dbCmd.AddCommand(removeCmd)
+	removeCmd.AddCommand(removeClustersCmd)
+	removeCmd.AddCommand(removeKPIsCmd)
+	removeCmd.AddCommand(removeErrorsCmd)
+
+	// Flags for 'remove clusters'
+	removeClustersCmd.Flags().StringVar(&removeClusterName, "name", "",
+		"cluster name to remove (required)")
+	_ = removeClustersCmd.MarkFlagRequired("name")
+
+	// Flags for 'remove kpis'
+	removeKPIsCmd.Flags().StringVar(&removeClusterName, "cluster-name", "",
+		"cluster name (required)")
+	removeKPIsCmd.Flags().StringVar(&removeKPIName, "name", "",
+		"KPI name to remove (optional)")
+	_ = removeKPIsCmd.MarkFlagRequired("cluster-name")
+
+	// Flags for 'remove errors'
+	removeErrorsCmd.Flags().StringVar(&removeKPIName, "name", "",
+		"KPI name to clear errors for")
+	removeErrorsCmd.Flags().BoolVar(&removeAll, "all", false,
+		"clear all error records")
+}
+
+func runRemoveClusters(cmd *cobra.Command, args []string) error {
+	db, _, err := connectToDB()
+	if err != nil {
+		return err
```

Comment:
```suggestion
		return fmt.Errorf("failed to connect to DB: %w", err)
```
This can be done in other places where you're calling the connectToDB(). Can be done in separate PR, though.